packages/convex-helpers/server/zod.ts

```diff
@@ -540,6 +540,18 @@ type ConvexUnionValidatorFromZod<T> = T extends z.ZodTypeAny[]
     >
   : never;
 
+type ConvexObjectValidatorFromZod<T extends Record<string, z.ZodTypeAny>> =
+  VObject<
+    ObjectType<{
+      [key in keyof T]: T[key] extends z.ZodTypeAny
+        ? ConvexValidatorFromZod<T[key]>
+        : never;
+    }>,
+    {
+      [key in keyof T]: ConvexValidatorFromZod<T[key]>;
```

Comment:
Why was the extends z.ZodTypeAny requiried in the above object type but not here?

Reply:
So tbh I kept pattern matching off of the `ConvexUnionValidatorFromZod` above until TS stopped complaining about a self referential type. Given the type bound on `T`, `T[key]` should always extend `z.ZodTypeAny`. I can do the same thing here too for consistency